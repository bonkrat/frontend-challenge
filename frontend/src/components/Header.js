import React from "react";
import * as styles from "./Header.styles";
import UserInfo from "./UserInfo";
import GlobalLoading from "../containers/GlobalLoading";
import { Link } from "react-router-dom";
import LinearProgress from "@material/react-linear-progress";

/**
 * Renders a sticky header at the top of the page.
 */
export default () => {
  const { isGlobalLoading } = GlobalLoading.useContainer();

  return (
    <div className={styles.root}>
      <div className={styles.header}>
        <Link to="/">
          <div className={styles.headerText}>Not Kickstarter</div>
        </Link>
        <UserInfo />
      </div>
      <LinearProgress
        indeterminate
        className={styles.loading(isGlobalLoading)}
      />
    </div>
  );
};
