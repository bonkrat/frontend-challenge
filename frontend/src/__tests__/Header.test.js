import React from "react";
import { shallow } from "enzyme";
import Header from "../components/Header";
import toJson from "enzyme-to-json";

jest.mock("../containers/GlobalLoading", () => ({
  useContainer: jest
    .fn()
    .mockReturnValueOnce({ isGlobalLoading: false })
    .mockReturnValueOnce({ isGlobalLoading: true })
}));

it("renders correctly", () => {
  expect(toJson(shallow(<Header />))).toMatchSnapshot();
});

it("renders loading styles correctly", () => {
  expect(toJson(shallow(<Header />))).toMatchSnapshot();
});
