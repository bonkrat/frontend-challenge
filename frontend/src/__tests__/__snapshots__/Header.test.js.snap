// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sticky top-0 flex flex-col z-10"
>
  <div
    className="flex px-16 border border-t-0 border-l-0 border-r-0 w-full justify-between bg-white"
  >
    <Link
      to="/"
    >
      <div
        className="font-bold h-16 flex justify-center items-center"
      >
        Not Kickstarter
      </div>
    </Link>
    <UserInfoQuery />
  </div>
  <t
    buffer={0}
    bufferingDots={true}
    className="-mt-1 opacity-0"
    closed={false}
    indeterminate={true}
    progress={0}
    reversed={false}
    tag="div"
  />
</div>
`;

exports[`renders loading styles correctly 1`] = `
<div
  className="sticky top-0 flex flex-col z-10"
>
  <div
    className="flex px-16 border border-t-0 border-l-0 border-r-0 w-full justify-between bg-white"
  >
    <Link
      to="/"
    >
      <div
        className="font-bold h-16 flex justify-center items-center"
      >
        Not Kickstarter
      </div>
    </Link>
    <UserInfoQuery />
  </div>
  <t
    buffer={0}
    bufferingDots={true}
    className="-mt-1 opacity-100"
    closed={false}
    indeterminate={true}
    progress={0}
    reversed={false}
    tag="div"
  />
</div>
`;
